### ISSUE TITLE
SC2 replay BC [4/6]: core BC fit + `--bc` entry point (default policy)

### ISSUE THREAD

**Issue by espenhk** [espenhk/gamer-ai#353]
Part of #349. Depends on [2/6].

## Goal
Fit a policy to the demonstration dataset and wire a standalone entry point that writes `policy_weights.yaml`, so the normal SC2 greedy/gradient loop fine-tunes from it.

## Scope
- **`fit_bc(dataset, obs_spec, *, target="sc2_reinforce", **opts)`** — **default target = MLP** (`TwoHeadREINFORCEPolicy`, `framework/reinforce.py:358`): supervised pre-train of the shared trunk + two heads — cross-entropy on the `fn_idx` softmax head, MSE on the sigmoid `x`/`y` head — via mini-batch gradient descent (`bc_epochs`, `bc_learning_rate`). Obs normalised with `obs_spec.scales`. Output loads via `TwoHeadREINFORCEPolicy.from_cfg` so a subsequent `sc2_reinforce` run resumes from it. A linear target (`sc2_genetic`, closed-form least-squares into `SC2MultiHeadLinearPolicy`) remains available via `--bc-target sc2_genetic`. (Remaining targets land in [5/6].)
- **`run(replay_dir, experiment_dir, obs_spec, **opts)`** — orchestrate read → fit → `policy.save(experiment_dir/policy_weights.yaml)` (+ trainer state where the target uses one); also write `bc_summary.json` (n_replays kept/skipped, n_episodes, n_pairs, fn_idx histogram, resolved player/race, final BC loss).
- **`main.py --bc` mode** (SC2-only, sibling to `--play`/`--eval` at `main.py:92-102`):
  `python main.py <name> --game sc2 --bc --replay-dir <dir> [--bc-player winner|1|2] [--bc-race terran|protoss|zerg|any] [--bc-target sc2_reinforce|sc2_genetic|...]`
  resolve experiment dir via `SC2Adapter.experiment_dir`, build obs_spec from `training_params.yaml`, run BC, then `python main.py <name> --game sc2` fine-tunes (loop auto-loads the existing weights/trainer files).
- **Defaults (decided)**: `--bc-player` → **`winner`**; `--bc-race` → **`any`**; `--bc-target` → **`sc2_reinforce`** (MLP).
- **Config keys** in `games/sc2/config/training_params.yaml` (documented): `bc_replay_dir`, `bc_player_id` (`winner`), `bc_race` (`any`), `bc_target` (`sc2_reinforce`), `bc_max_replays`, `bc_step_mul` (= `step_mul`), `bc_epochs`, `bc_learning_rate`, `bc_batch_size`, `bc_ignore_noop` (`true`). CLI flags override config.

## Edge cases
- `bc_ignore_noop` drops idle steps so the fn head isn't swamped.
- Non-spatial actions masked out of the spatial (x/y) loss term.
- `--bc` rejected (SystemExit) for non-SC2 games (mirror `main.py:202-205`).
- Race filter / empty dataset → fail fast with a clear message (no empty weight file).

## Acceptance criteria
- `fit_bc` (MLP) on a separable synthetic dataset drives BC loss down and recovers correct `fn_idx`/coords on held-out steps; saved weights load + run as `sc2_reinforce`.
- `--bc-target sc2_genetic` still produces a loadable `SC2MultiHeadLinearPolicy` champion.
- `run` writes valid `policy_weights.yaml` (+ trainer state) + `bc_summary.json`; CLI guard, winner-default, and race-filter passthrough tested.

## Validation
`python -m pytest tests/test_sc2_replay_bc.py`. Update `tests/README.md`.
